addons/website_sale/controllers/main.py

```diff
@@ -577,28 +575,38 @@ def resequence_product_image(self, image_res_model, image_res_id, move):
         # no-op resequences
         if new_image_idx == image_idx:
             return
-        # We can not move an embedded image to the first position (main product image)
-        if image_res_model == 'product.image' and image_to_resequence.video_url and product_images[new_image_idx]._name != 'product.image':
-            raise ValidationError(_("Can not resequence embedded image/video with a non compatible image."))
-
-        # Swap images
-        other_image = product_images[new_image_idx]
-        source_field = hasattr(image_to_resequence, 'video_url') and image_to_resequence.video_url and 'video_url' or 'image_1920'
-        target_field = hasattr(other_image, 'video_url') and other_image.video_url and 'video_url' or 'image_1920'
-        if target_field == 'video_url' and image_res_model == 'product.product':
-            raise ValidationError(_("Can not resequence a video at first position."))
-        previous_data = other_image[target_field]
-        other_image[source_field] = image_to_resequence[source_field]
-        image_to_resequence[target_field] = previous_data
-        if source_field == 'video_url' and target_field != 'video_url':
-            image_to_resequence.video_url = False
-        if target_field == 'video_url' and source_field != 'video_url':
-            other_image.video_url = False
-
-        if hasattr(other_image, 'video_url'):
-            other_image._onchange_video_url()
-        if hasattr(image_to_resequence, 'video_url'):
-            image_to_resequence._onchange_video_url()
+
+        # Reorder images locally.
+        product_images.insert(new_image_idx, product_images.pop(image_idx))
+
+        # If the main image has been reordered (i.e. it's no longer in first position), use the
+        # image that's now in first position as main image instead.
+        main_image_idx = 0
+        for idx, product_image in enumerate(product_images):
+            # Additional images are product.image records. The main image is a product.product or
+            # product.template record.
+            if product_image._name != 'product.image':
+                main_image_idx = idx
+                break
```

Comment:
I don't mind doing this, but this will systematically look at all images, which can be a bit slower. Also, at this point in the code, there should always be a main image, so we don't need the default, and we could simply write `[idx for idx, image in enumerate(product_images) if image._name != 'product.image'][0]`. WDYT?

Reply:
As far as I know `next` returns the first element that matches the condition, so it is equivalent of `for` loop with `break`. In your suggestion it will always go through the whole list.
The default can be removed if there is always a main image ofc.

Reply:
Oh, so the first argument in the next function is an iterator? I thought you were passing a list (evaluated before applying next). Either way, it doesn't change the complexity so I'll update it.